cmd/manager/operator.go

```diff
@@ -293,8 +293,16 @@ func RunOperator(cmd *cobra.Command, args []string) {
 		setupLog.Error(err, "")
 		os.Exit(1)
 	}
-	pflag, _ := flags.GetString("platform")
-	platform := getValidPlatform(pflag)
+
+	pEnv := os.Getenv("PLATFORM")
+	// We respect the environment variable over the flag
+	// This is to support the operator being deployed by OLM
+	if pEnv == "" {
+		pflag, _ := flags.GetString("platform")
+		os.Setenv("PLATFORM", pflag)
+	}
+	pEnv = os.Getenv("PLATFORM")
```

Comment:
why do we set and then get the platform variable? Why not just use a local variable?